src/run.ts

```diff
@@ -168,13 +175,83 @@ const requireChangesetsCliPkgJson = (cwd: string) => {
   }
 };
 
+type GetMessageOptions = {
+  hasPublishScript: boolean;
+  branch: string;
+  changedPackagesInfo: {
+    highestLevel: number;
+    private: boolean;
+    content: string;
+    header: string;
+  }[];
+  maxCharactersPerMessage: number;
+  preState?: PreState;
+};
+
+export async function getChangesetsMessage({
+  hasPublishScript,
+  preState,
+  changedPackagesInfo,
+  maxCharactersPerMessage,
+  branch,
+}: GetMessageOptions) {
+  let messageHeader = `This PR was opened by the [Changesets release](https://github.com/changesets/action) GitHub action. When you're ready to do a release, you can merge this and ${
+    hasPublishScript
+      ? `the packages will be published to npm automatically`
+      : `publish to npm yourself or [setup this action to publish automatically](https://github.com/changesets/action#with-publishing)`
+  }. If you're not ready to do a release yet, that's fine, whenever you add more changesets to ${branch}, this PR will be updated.
+`;
+  let messagePrestate = !!preState
+    ? `⚠️⚠️⚠️⚠️⚠️⚠️
+
+\`${branch}\` is currently in **pre mode** so this branch has prereleases rather than normal releases. If you want to exit prereleases, run \`changeset pre exit\` on \`${branch}\`.
+
+⚠️⚠️⚠️⚠️⚠️⚠️
+`
+    : "";
+  let messageReleases = `# Releases`;
+
+  let fullMessage = [
+    messageHeader,
+    messagePrestate,
+    messageReleases,
+    ...changedPackagesInfo.map((info) => `${info.header}\n\n${info.content}`),
+  ].join("\n");
+
+  // Check that the message does not exceed the size limit.
+  // If not, omit the changelog entries of each package.
+  if (fullMessage.length > maxCharactersPerMessage) {
+    fullMessage = [
+      messageHeader,
+      messagePrestate,
+      messageReleases,
+      `\n> The changelog information of each package has been omitted from this message, as the content exceeds the size limit.\n`,
+      ...changedPackagesInfo.map((info) => `${info.header}\n\n`),
+    ].join("\n");
+  }
+
+  // Check (again) that the message is within the size limit.
+  // If not, omit all release content this time.
+  if (fullMessage.length > maxCharactersPerMessage) {
```

Comment:
heh, this is probably very, very unlikely :P but I appreciate thinking about this edge case 👍